src/main/java/me/totalfreedom/totalfreedommod/command/Command_coins.java

```diff
@@ -21,34 +24,119 @@ public boolean run(CommandSender sender, Player playerSender, Command cmd, Strin
             msg("The shop is currently disabled!", ChatColor.RED);
             return true;
         }
-        Player p;
+
         final String prefix = FUtil.colorize(ConfigEntry.SHOP_PREFIX.getString() + " ");
-        if (args.length > 0)
+
+        switch (args.length)
         {
-            if (getPlayer(args[0]) != null)
+            // Mode for seeing how many coins the sender has (doesn't work from console)
+            case 0:
             {
-                p = getPlayer(args[0]);
+                if (senderIsConsole)
+                {
+                    msg("When used from the console, you must define a target player.");
+                }
+                else
+                {
+                    PlayerData playerData = getData(playerSender);
+                    msg(prefix + ChatColor.GREEN + "You have " + ChatColor.RED + playerData.getCoins() + ChatColor.GREEN
+                            + " coins.");
+                }
+                return true;
             }
-            else
+
+            // Mode for seeing how many coins a player has.
+            case 1:
             {
-                msg(PLAYER_NOT_FOUND);
+                Player target = getPlayer(args[0]);
+
+                if (target == null)
+                {
+                    msg(PLAYER_NOT_FOUND);
+                }
+                else
+                {
+                    PlayerData playerData = getData(target);
+                    msg(prefix + ChatColor.GREEN + target.getName() + " has " + ChatColor.RED + playerData.getCoins() + ChatColor.GREEN + " coins.");
+                }
                 return true;
             }
-        }
-        else
-        {
-            if (senderIsConsole)
+
+            // Mode for paying another player coins
+            case 3:
             {
-                msg(prefix + ChatColor.RED + "You are not a player, use /coins <playername>");
-                return true;
+                if (args[0].equalsIgnoreCase("pay"))
```

Comment:
Do we catch the else anywhere to basically throw a not found command / similar?

Reply:
Yes; it returns false at line 121